### ISSUE TITLE
RuntimeError: CA configuration failed.

### ISSUE THREAD

**Issue by bittner** [painless-software/groundcontrol#1]
As of now, the Groundcontrol Genesis system is a Vagrant box hosting the various components. Current status (see the [Ansible playbook](https://github.com/painless-software/groundcontrol/blob/master/ansible/playbook.yml) for context):

- Installing `virtualization` (libvirt) works fine (result untested, however)
- Installing `encfrontend` alone (w/ or w/o libvirt) works fine
- Installing `identitymanagement` alone (w/ or w/o libvirt) works fine
- Installing both `encfrontend` and `identitymanagement` **leads to a conflict**

Details
---------
```
  Configuring NTP daemon (ntpd)
  Configuring directory server (dirsrv). Estimated time: 30 seconds
  Configuring Kerberos KDC (krb5kdc)
  Configuring kadmin
  Configuring certificate server (pki-tomcatd). Estimated time: 3 minutes
    [error] RuntimeError: CA configuration failed.
```

### FILE: `/var/log/ipaserver-install.log` (relevant parts)

```python
2018-03-30T14:25:18Z DEBUG Starting external process
2018-03-30T14:25:18Z DEBUG args=/usr/sbin/pkispawn -s CA -f /tmp/tmpub3ub7
2018-03-30T14:26:22Z DEBUG Process finished, return code=1
2018-03-30T14:26:22Z DEBUG stdout=Log file: /var/log/pki/pki-ca-spawn.20180330142518.log
Loading deployment configuration from /tmp/tmpub3ub7.
Installing CA into /var/lib/pki/pki-tomcat.
Storing deployment configuration into /etc/sysconfig/pki/tomcat/pki-tomcat/ca/deployment.cfg.

Installation failed: server failed to restart

2018-03-30T14:26:22Z DEBUG stderr=pkispawn    : ERROR    ....... server failed to restart

2018-03-30T14:26:22Z CRITICAL Failed to configure CA instance: Command '/usr/sbin/pkispawn -s CA -f /tmp/tmpub3ub7' returned non-zero exit status 1
2018-03-30T14:26:22Z CRITICAL See the installation logs and the following files/directories for more information:
2018-03-30T14:26:22Z CRITICAL   /var/log/pki/pki-tomcat
2018-03-30T14:26:22Z DEBUG Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/ipaserver/install/service.py", line 504, in start_creation
    run_step(full_msg, method)
  File "/usr/lib/python2.7/site-packages/ipaserver/install/service.py", line 494, in run_step
    method()
  File "/usr/lib/python2.7/site-packages/ipaserver/install/cainstance.py", line 615, in __spawn_instance
    self.tmp_agent_pwd)
  File "/usr/lib/python2.7/site-packages/ipaserver/install/dogtaginstance.py", line 148, in spawn_instance
    self.handle_setup_error(e)
  File "/usr/lib/python2.7/site-packages/ipaserver/install/dogtaginstance.py", line 398, in handle_setup_error
    raise RuntimeError("%s configuration failed." % self.subsystem)
RuntimeError: CA configuration failed.

2018-03-30T14:26:22Z DEBUG   [error] RuntimeError: CA configuration failed.
2018-03-30T14:26:22Z DEBUG   File "/usr/lib/python2.7/site-packages/ipapython/admintool.py", line 172, in execute
    return_value = self.run()
  File "/usr/lib/python2.7/site-packages/ipapython/install/cli.py", line 333, in run
    cfgr.run()
  File "/usr/lib/python2.7/site-packages/ipapython/install/core.py", line 368, in run
    self.execute()
  File "/usr/lib/python2.7/site-packages/ipapython/install/core.py", line 392, in execute
    for _nothing in self._executor():
  File "/usr/lib/python2.7/site-packages/ipapython/install/core.py", line 434, in __runner
    exc_handler(exc_info)
  File "/usr/lib/python2.7/site-packages/ipapython/install/core.py", line 463, in _handle_execute_exception
    self._handle_exception(exc_info)
  File "/usr/lib/python2.7/site-packages/ipapython/install/core.py", line 453, in _handle_exception
    six.reraise(*exc_info)
  File "/usr/lib/python2.7/site-packages/ipapython/install/core.py", line 424, in __runner
    step()
  File "/usr/lib/python2.7/site-packages/ipapython/install/core.py", line 421, in <lambda>
    step = lambda: next(self.__gen)
  File "/usr/lib/python2.7/site-packages/ipapython/install/util.py", line 81, in run_generator_with_yield_from
    six.reraise(*exc_info)
  File "/usr/lib/python2.7/site-packages/ipapython/install/util.py", line 59, in run_generator_with_yield_from
    value = gen.send(prev_value)
  File "/usr/lib/python2.7/site-packages/ipapython/install/core.py", line 658, in _configure
    next(executor)
  File "/usr/lib/python2.7/site-packages/ipapython/install/core.py", line 434, in __runner
    exc_handler(exc_info)
  File "/usr/lib/python2.7/site-packages/ipapython/install/core.py", line 463, in _handle_execute_exception
    self._handle_exception(exc_info)
  File "/usr/lib/python2.7/site-packages/ipapython/install/core.py", line 521, in _handle_exception
    self.__parent._handle_exception(exc_info)
  File "/usr/lib/python2.7/site-packages/ipapython/install/core.py", line 453, in _handle_exception
    six.reraise(*exc_info)
  File "/usr/lib/python2.7/site-packages/ipapython/install/core.py", line 518, in _handle_exception
    super(ComponentBase, self)._handle_exception(exc_info)
  File "/usr/lib/python2.7/site-packages/ipapython/install/core.py", line 453, in _handle_exception
    six.reraise(*exc_info)
  File "/usr/lib/python2.7/site-packages/ipapython/install/core.py", line 424, in __runner
    step()
  File "/usr/lib/python2.7/site-packages/ipapython/install/core.py", line 421, in <lambda>
    step = lambda: next(self.__gen)
  File "/usr/lib/python2.7/site-packages/ipapython/install/util.py", line 81, in run_generator_with_yield_from
    six.reraise(*exc_info)
  File "/usr/lib/python2.7/site-packages/ipapython/install/util.py", line 59, in run_generator_with_yield_from
    value = gen.send(prev_value)
  File "/usr/lib/python2.7/site-packages/ipapython/install/common.py", line 63, in _install
    for _nothing in self._installer(self.parent):
  File "/usr/lib/python2.7/site-packages/ipaserver/install/server/__init__.py", line 578, in main
    master_install(self)
  File "/usr/lib/python2.7/site-packages/ipaserver/install/server/install.py", line 248, in decorated
    func(installer)
  File "/usr/lib/python2.7/site-packages/ipaserver/install/server/install.py", line 798, in install
    ca.install_step_0(False, None, options)
  File "/usr/lib/python2.7/site-packages/ipaserver/install/ca.py", line 284, in install_step_0
    use_ldaps=standalone)
  File "/usr/lib/python2.7/site-packages/ipaserver/install/cainstance.py", line 447, in configure_instance
    self.start_creation(runtime=runtime)
  File "/usr/lib/python2.7/site-packages/ipaserver/install/service.py", line 504, in start_creation
    run_step(full_msg, method)
  File "/usr/lib/python2.7/site-packages/ipaserver/install/service.py", line 494, in run_step
    method()
  File "/usr/lib/python2.7/site-packages/ipaserver/install/cainstance.py", line 615, in __spawn_instance
    self.tmp_agent_pwd)
  File "/usr/lib/python2.7/site-packages/ipaserver/install/dogtaginstance.py", line 148, in spawn_instance
    self.handle_setup_error(e)
  File "/usr/lib/python2.7/site-packages/ipaserver/install/dogtaginstance.py", line 398, in handle_setup_error
    raise RuntimeError("%s configuration failed." % self.subsystem)

2018-03-30T14:26:22Z DEBUG The ipa-server-install command failed, exception: RuntimeError: CA configuration failed.
2018-03-30T14:26:22Z ERROR CA configuration failed.
2018-03-30T14:26:22Z ERROR The ipa-server-install command failed. See /var/log/ipaserver-install.log for more information
```

**Comment by bittner** [comment]
I found this in the logfile: (`/var/log/ipaserver-install.log`)

```python
2018-03-31T17:53:03Z DEBUG Starting external process
2018-03-31T17:53:03Z DEBUG args=/usr/bin/certutil -d /etc/dirsrv/slapd-PAINLESS-SOFTWARE/ -L -n PAINLESS.SOFTWARE IPA CA -a -f /etc/dirsrv/slapd-PAINLESS-SOFTWARE/pwdfile.txt
2018-03-31T17:53:03Z DEBUG Process finished, return code=255
2018-03-31T17:53:03Z DEBUG stdout=
2018-03-31T17:53:03Z DEBUG stderr=certutil: Could not find cert: PAINLESS.SOFTWARE IPA CA
: PR_FILE_NOT_FOUND_ERROR: File not found
```

and

```python
2018-03-31T17:55:09Z DEBUG Starting external process
2018-03-31T17:55:09Z DEBUG args=/usr/bin/certutil -d /etc/httpd/alias -L -n ipaCert -a -f /etc/httpd/alias/pwdfile.txt
2018-03-31T17:55:10Z DEBUG Process finished, return code=255
2018-03-31T17:55:10Z DEBUG stdout=
2018-03-31T17:55:10Z DEBUG stderr=certutil: Could not find cert: ipaCert
: PR_FILE_NOT_FOUND_ERROR: File not found
```

and

```python
2018-03-31T17:57:12Z DEBUG The DNS query name does not exist: groundcontrol.painless.software.
2018-03-31T17:57:17Z ERROR unable to resolve host name groundcontrol.painless.software. to IP address, ipa-ca DNS record will be incomplete
```

Here are links to the man pages of the commands in use:

- [ipa-server-install](https://www.mankier.com/1/ipa-server-install#Options-Certificate_System_Options)
- [pkispawn](https://www.mankier.com/8/pkispawn)

**Comment by bittner** [comment]
Fixed via 369d3f5138d9af3c4d4a5e4120c86a874e1288b2.

Instead of trying to make the two softwares collaborate we run them in separate VMs.

**Reading suggestion:** [Multiple Vagrant VMs in One Vagrantfile](http://www.thisprogrammingthing.com/2015/multiple-vagrant-vms-in-one-vagrantfile/)
